SYCL/Reduction/reduction_nd_N_vars.cpp

```diff
@@ -17,188 +17,170 @@ constexpr access::mode RW = access::mode::read_write;
 constexpr access::mode DW = access::mode::discard_write;
 
 template <typename RangeT>
-void printNVarsTestLabel(bool IsSYCL2020, const RangeT &Range,
-                         bool ToCERR = false) {
+void printNVarsTestLabel(const RangeT &Range, bool ToCERR = false) {
   std::ostream &OS = ToCERR ? std::cerr : std::cout;
-  std::string Mode = IsSYCL2020 ? "SYCL2020" : "ext::oneapi  ";
-  OS << (ToCERR ? "Error" : "Start") << ": Mode=" << Mode
-     << ", Range=" << Range;
+  OS << (ToCERR ? "Error" : "Start") << ", Range=" << Range;
   if (!ToCERR)
     OS << std::endl;
 }
 
-// Returns 0 if the test case passed. Otherwise, some non-zero value.
-template <class Name, bool IsSYCL2020, typename T1, access::mode Mode1,
-          typename T2, access::mode Mode2, typename T3, access::mode Mode3,
-          typename T4, access::mode Mode4, class BinaryOperation1,
-          class BinaryOperation2, class BinaryOperation3,
-          class BinaryOperation4>
-int testOne(queue &Q, T1 IdentityVal1, T1 InitVal1, BinaryOperation1 BOp1,
-            T2 IdentityVal2, T2 InitVal2, BinaryOperation2 BOp2,
-            T3 IdentityVal3, T3 InitVal3, BinaryOperation3 BOp3,
-            T4 IdentityVal4, T3 InitVal4, BinaryOperation4 BOp4,
-            usm::alloc AllocType4, size_t NWorkItems, size_t WGSize) {
+template <typename T, bool IsUsm, usm::alloc AllocType> auto InitOut(queue &Q) {
+  if constexpr (IsUsm)
+    return (T *)malloc(1, Q, AllocType);
```

Comment:
IMO, it is good to stay on safer side and keep the checks like "Dev.get_info<info::device::usm_device_allocations>()" before using malloc.

Also, it looks like a mistake here. You alloc too few memory - 1 byte only.
If you want to alloc 1 element, then:
```suggestion
    return malloc<T>(1, Q, AllocType);
```